nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-headless-server/src/main/java/org/apache/nifi/headless/HeadlessNiFiServer.java

```diff
@@ -215,6 +228,14 @@ public StatusHistoryDumpFactory getStatusHistoryDumpFactory() {
     public void stop() {
         try {
             flowService.stop(false);
+
+            try {
+                if (narAutoLoader != null) {
+                    narAutoLoader.stop();
```

Comment:
It looks like narAutoLoader is being stopped twice which results in a logged exception. As discussed could you please take a look?
```
2022-02-24 18:54:37,969 INFO [pool-2-thread-2] org.apache.nifi.BootstrapListener Received SHUTDOWN request from Bootstrap
2022-02-24 18:54:37,970 INFO [pool-2-thread-2] org.apache.nifi.minifi.MiNiFi Initiating shutdown of MiNiFi server...
2022-02-24 18:54:37,971 INFO [pool-2-thread-2] o.apache.nifi.controller.FlowController Initiated graceful shutdown of flow controller...waiting up to 10 seconds
2022-02-24 18:54:38,780 INFO [pool-2-thread-2] o.apache.nifi.controller.FlowController Controller has been terminated successfully.
2022-02-24 18:54:38,792 INFO [pool-2-thread-2] org.apache.nifi.nar.NarAutoLoaderTask Stopping NAR Auto-loader
2022-02-24 18:54:38,794 INFO [pool-2-thread-2] org.apache.nifi.nar.NarAutoLoader NAR Auto-Loader stopped
2022-02-24 18:54:38,796 INFO [pool-2-thread-2] org.apache.nifi.BootstrapListener Successfully initiated communication with Bootstrap
2022-02-24 18:54:38,796 INFO [pool-2-thread-2] org.apache.nifi.minifi.MiNiFi MiNiFi server shutdown completed (nicely or otherwise).
2022-02-24 18:54:38,797 INFO [Thread-1] org.apache.nifi.minifi.MiNiFi Initiating shutdown of MiNiFi server...
2022-02-24 18:54:38,799 WARN [Thread-1] o.a.nifi.headless.HeadlessNiFiServer Failed to stop NAR auto-loader
java.lang.NullPointerException: null
	at org.apache.nifi.nar.NarAutoLoader.stop(NarAutoLoader.java:113)
	at org.apache.nifi.headless.HeadlessNiFiServer.stop(HeadlessNiFiServer.java:234)
	at org.apache.nifi.minifi.MiNiFi.shutdownHook(MiNiFi.java:183)
	at org.apache.nifi.minifi.MiNiFi.lambda$new$1(MiNiFi.java:88)
	at java.lang.Thread.run(Thread.java:748)
```

Reply:
Good catch, I'll look into who is closing it before HeadlessNiFiServer, but will also put in some defensive code so you can call stop() more than once without getting an NPE.

Reply:
I rebuilt and re-validated with the latest. Tried with empty and prepopulated ./extensins folder too and with both approaches it worked and I didn't see the exception so working as intended :) Thanks for the change
```
2022-02-25 20:32:04,870 INFO [NAR Auto-Loader] org.apache.nifi.nar.NarAutoLoaderTask Found ./extensions/nifi-hl7-nar-1.16.0-SNAPSHOT.nar in auto-load directory
2022-02-25 20:32:04,870 INFO [NAR Auto-Loader] org.apache.nifi.nar.StandardNarLoader Starting load process for 1 NARs...
2022-02-25 20:32:05,078 INFO [NAR Auto-Loader] org.apache.nifi.nar.StandardNarLoader Creating class loaders for 1 NARs...
2022-02-25 20:32:05,080 INFO [NAR Auto-Loader] org.apache.nifi.nar.NarClassLoaders Loaded NAR file: /Users/cbejan/Work/Cloudera/git/nifi/minifi/minifi-assembly/target/minifi-1.16.0-SNAPSHOT-bin/minifi-1.16.0-SNAPSHOT/./work/nar/extensions/nifi-hl7-nar-1.16.0-SNAPSHOT.nar-unpacked as class loader org.apache.nifi.nar.NarClassLoader[./work/nar/extensions/nifi-hl7-nar-1.16.0-SNAPSHOT.nar-unpacked]
2022-02-25 20:32:05,080 INFO [NAR Auto-Loader] org.apache.nifi.nar.StandardNarLoader Successfully created class loaders for 1 NARs, 0 were skipped
2022-02-25 20:32:05,086 INFO [NAR Auto-Loader] o.a.n.n.StandardExtensionDiscoveringManager Loaded extensions for org.apache.nifi:nifi-hl7-nar:1.16.0-SNAPSHOT in 6 millis
2022-02-25 20:32:05,145 INFO [NAR Auto-Loader] org.apache.nifi.nar.StandardNarLoader Finished NAR loading process!
```